File: 2270. Number of Ways to Split Array/main.go
package main

func sum(nums []int) int {
	res := 0
	for _, n := range nums {
		res += n
	}
	return res
}

func waysToSplitArray(nums []int) int {
	total := sum(nums)
	res := 0
	cur := 0

	for _, n := range nums[:len(nums)-1] {
		cur += n
		if cur >= total-cur {
			res++
		}
	}

	return res
}

File: 2270. Number of Ways to Split Array/main.py
from typing import List


class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        total = sum(nums)
        cur, res = 0, 0
        
        for n in nums[:-1]:
            cur += n
            if cur >= total - cur:
                res += 1

        return res
